src/bootstrap/bin/main.rs

```diff
@@ -12,5 +12,31 @@ use bootstrap::{Build, Config};
 fn main() {
     let args = env::args().skip(1).collect::<Vec<_>>();
     let config = Config::parse(&args);
+    check_version(&config);
     Build::new(config).build();
 }
+
+fn check_version(config: &Config) {
+    const VERSION: usize = 0;
+
+    let version_suggestion = if let Some(version) = config.version {
```

Comment:
Why should we allocate a `String` here when we don't need to? We could just duplicate the print here right?

Reply:
This is a very tiny fraction of the runtime ... just printing it will be 100x as slow as the allocation.